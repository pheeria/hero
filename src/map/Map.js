import React, { useState } from 'react';
import {
  withGoogleMap,
  GoogleMap,
  Marker,
  OverlayView
} from 'react-google-maps';
import { Card } from 'antd';
import Adder from '../adder/Adder';

const Map = withGoogleMap(
  ({ places, preset, berlin, addPlace, presetLocation }) => {
    const [center, setCenter] = useState(preset || berlin);

    return (
      <GoogleMap defaultZoom={14} center={center}>
        {preset ? (
          <OverlayView
            position={preset}
            mapPaneName={OverlayView.OVERLAY_MOUSE_TARGET}
          >
            <Card>
              <Adder
                location={preset}
                addPlace={newPlace => {
                  addPlace(newPlace);
                  presetLocation(null);
                }}
                reset={() => presetLocation(null)}
              />
            </Card>
          </OverlayView>
        ) : (
          places.map(place => (
            <Marker
              onClick={() => setCenter(place.location)}
              key={place.id}
              position={place.location}
            />
          ))
        )}
      </GoogleMap>
    );
  }
);

export default Map;
