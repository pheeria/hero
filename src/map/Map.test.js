import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Map from './Map';

jest.spyOn(React, 'useState').mockImplementation(v => [v, jest.fn()]);

describe('Map', () => {
  it('should render map container without crashing', () => {
    const tempLocation = {
      lat: 52.47732379999999,
      lng: 13.425160300000016
    };

    const wrapper = shallow(
      <Map
        tempLocation={tempLocation}
        places={[]}
        selectedPlace={tempLocation}
        city={tempLocation}
        loadingElement={<div style={{ height: `100%` }} />}
        containerElement={<div style={{ height: `calc(100vh - 64px)` }} />}
        mapElement={<div style={{ height: `100%` }} />}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
