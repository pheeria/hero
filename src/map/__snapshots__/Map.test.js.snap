// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map should render map container without crashing 1`] = `
<div
  style={
    Object {
      "height": "calc(100vh - 64px)",
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
      }
    }
  />
  <div />
</div>
`;
